### PR TITLE
Implement youtube subscription import from Google takeout

### PR DESCRIPTION
**Pull request by Stypox**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [x] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

This PR replaces the current approach of importing subscriptions (since it has been removed by YouTube) with Google takeout. Fixes TeamNewPipe/NewPipe#4720
I implemented a test with a real-world Google takeout file and it worked flawlessy.

Depends on TeamNewPipe/NewPipe#4759

### CODE REVIEW

**Review by TobiGr — COMMENTED** [review]
Looks good

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSubscriptionExtractor.java [hunk 1]
```diff
@@ -1,126 +1,71 @@
...
-    public YoutubeSubscriptionExtractor(YoutubeService service) {
-        super(service, Collections.singletonList(INPUT_STREAM));
+    public YoutubeSubscriptionExtractor(final YoutubeService youtubeService) {
+        super(youtubeService, Collections.singletonList(INPUT_STREAM));
     }
 
     @Override
     public String getRelatedUrl() {
-        return "https://www.youtube.com/subscription_manager?action_takeout=1";
+        return "https://takeout.google.com/takeout/custom/youtube?continue=https%3A%2F%2Fmyaccount.google.com%2Fyourdata%2Fyoutube%3Fhl%3Den_GB%26authuser%3D0";
```
[the remark this answers is not in the document]
Comment (simula67) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSubscriptionExtractor.java:34]: URL decoder at https://www.urldecoder.org/ shows that

https://takeout.google.com/takeout/custom/youtube?continue=https%3A%2F%2Fmyaccount.google.com%2Fyourdata%2Fyoutube%3Fhl%3Den_GB%26authuser%3D0

to be

 https://takeout.google.com/takeout/custom/youtube?continue=https://myaccount.google.com/yourdata/youtube?hl=en_GB&authuser=0

Removing the URL arguments:

https://takeout.google.com/takeout/custom/youtube?continue=https%3A%2F%2Fmyaccount.google.com%2Fyourdata%2Fyoutube

is

https://takeout.google.com/takeout/custom/youtube?continue=https://myaccount.google.com/yourdata/youtube

This seems to work too for downloading the subscriptions

[the remark this answers is not in the document]
Comment (Stypox) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSubscriptionExtractor.java:34]: I just used the url from TeamNewPipe/NewPipe#4759, but https://takeout.google.com/takeout/custom/youtube?continue=https://myaccount.google.com/yourdata/youtube is shorter and more human-readable, so I'm gonna use that

[the remark this answers is not in the document]
Comment (Stypox) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSubscriptionExtractor.java:34]: Oh, it even works without any `continue`: https://takeout.google.com/takeout/custom/youtube

[the remark this answers is not in the document]
Comment (Stypox) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSubscriptionExtractor.java]: Done

**Review by TobiGr — APPROVED** [review]

**Comment by B0pol** [comment]
Regression: cannot import subscriptions from invidious anymore

**Comment by snotrman** [comment]
Hey.. Just a kind reminder to update the version on FDroid, because you got a promo on https://www.youtube.com/watch?v=ED_Yh6nsGaQ
which is the way I found out about you, but the version on FDroid still says to get the subscriptions from youtube.com/subscription_manager?action_takeout.. which doesn't work.
It took me a while to realize that I don't have the latest version.

**Pull request merged**